### PR TITLE
Sensor device class docs

### PR DESCRIPTION
**Pull request by OttoWinter**
**Description:**

To make the frontend icon package a lot smaller, sensors should in most cases now report their icon via `device_class`(if the device class is common). See https://github.com/home-assistant/home-assistant/pull/14010, https://github.com/home-assistant/home-assistant-polymer/pull/1115 and https://github.com/home-assistant/home-assistant-polymer/pull/1100

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#14010

## Device Class Icons Screenshot

The page includes a screenshot of all supported device classes. If new device classes are added, this screenshot should also be updated. Use this configuration and setup the new device class template sensors to do so:

```yaml
sensor:
  - platform: template
    sensors:
      empty:
        friendly_name: Empty device_class
        value_template: 0
      battery:
        friendly_name: device_class battery
        value_template: 70
        unit_of_measurement: '%'
        device_class: battery
      humidity:
        friendly_name: device_class humidity
        value_template: 77
        unit_of_measurement: '%'
        device_class: humidity
      temperature:
        friendly_name: device_class temperature
        value_template: 12.2
        unit_of_measurement: °C
        device_class: temperature

group:
  sensor_device_classes:
    name: Sensor Device Classes
    entities:
      - sensor.empty
      - sensor.battery
      - sensor.humidity
      - sensor.temperature
```

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/sensor.markdown [hunk 1]
```diff
@@ -9,8 +9,15 @@ sharing: true
 footer: true
 ---
 
-Sensors are gathering information about states and conditions. 
+Sensors are gathering information about states and conditions.
 
 Home Assistant currently supports a wide range of sensors. They are able to display information which are provides by Home Assistant directly, are gathered from web services, and, of course, physical devices. Please check the sidebar for a full list of supported sensor platforms.
 
+### {% linkable_title Device Class %}
```
Comment (OttoWinter) [inline source/_components/sensor.markdown:16]: The configuration variables generator is pointing `device_class` options to `/components/%{component}/#device_class`, so there needs to be a title here for the link to work.

https://github.com/home-assistant/home-assistant.github.io/blob/7bfe31cbac81b23764066ed41a8d988384a78f01/plugins/configuration.rb#L5

File: source/_docs/configuration/customizing-devices.markdown [hunk 2]
```diff
@@ -71,9 +71,10 @@ homeassistant:
 
 ### {% linkable_title Device Class %}
 
-Device class is currently supported by the following platforms:
+Device class is currently supported by the following domains:
```
Comment (OttoWinter) [inline source/_docs/configuration/customizing-devices.markdown]: Just wondering: I do understand the difference between platforms (`sensor.mqtt` for example) and components (`sensor`) - or at least I think I do.

But I've also heard the term "domain" a couple of times. Is a domain the same thing as component or rather the name of a component? Which one is preferred to be used?

Comment (fabaff) [reply source/_docs/configuration/customizing-devices.markdown]: I think that we should avoid using domain here. In the user documentation we should stick to "component" and "platform".

**Review by fabaff — APPROVED** [review]
:bird:

**Pull request merged**